commands/validators.js

```diff
@@ -0,0 +1,114 @@
+const exitOnError = require('../utils/exit-on-error');
+const connect = require('../utils/connect');
+const { utils } = require('near-api-js');
+const BN = require('bn.js');
+const AsciiTable = require('ascii-table');
+
+function findSeatPrice(validators, numSeats) {
+    let stakes = validators.map((v) => new BN(v.stake, 10)).sort((a, b) => a.cmp(b));
+    let num = new BN(numSeats);
+    let stakesSum = stakes.reduce((a, b) => a.add(b));
+    // assert stakesSum >= numSeats
+    let left = new BN(1), right = stakesSum.add(new BN(1));
+    while (!left.eq(right.sub(new BN(1)))) {
+        const mid = left.add(right).div(new BN(2));
+        let found = false;
+        let currentSum = new BN(0);
+        for (let i = 0; i < stakes.length; ++i) {
+            currentSum = currentSum.add(stakes[i].div(mid));
+            if (currentSum.gte(num)) {
+                left = mid;
+                found = true;
+                break;
+            }
+        }
+        if (!found) {
+            right = mid;
+        }
+    }
+    return left;
+}
+
+function diffValidators(currentValidators, nextValidators) {
+    const result = { newValidators: [], removedValidators: [], changeValidators: [] };
+    const validatorsMap = new Map();
+    const nextValidatorsMap = new Map();
+    nextValidators.map((validator) => nextValidatorsMap[validator.account_id] = validator);
+    currentValidators.map((validator) => { 
+        if (!(validator.account_id in nextValidatorsMap)) {
+            result.removedValidators.push(validator);
+        }
+        validatorsMap[validator.account_id] = validator;
+    });
+    nextValidators.map((validator) => {
+        if (validator.account_id in validatorsMap) {
+            if (validatorsMap[validator.account_id].stake != validator.stake) {
+                result.changeValidators.push({new: validator, old: validatorsMap[validator.account_id]});
+            }
+        } else {
+            result.newValidators.push(validator);
+        }
+    });
+    return result;
+}
+
+module.exports = {
+    command: 'EXPERIMENTAL_validators',
```

Comment:
seems like a bit suboptimal to prefix commands with allcaps `EXPERIMENTAL_` thing. How about we just use `_` as prefix (similar to "private" variables / methods in JS).

Reply:
hm, was following our RPC standard for experimental APIs by @frol .
What is the common way for experimental CLI flags?

Reply:
It is common to have `experimental` prefix for CLI arguments (well, it is usually in lowercase):

* Node.js: https://nodejs.org/api/cli.html#cli_experimental_import_meta_resolve
* Kubectl: https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/

Docker has special config and env variable to enable experimental CLI arguments: https://github.com/docker/docker.github.io/issues/5718

P.S. it is standard to have kebab-case naming for the CLI arguments (for some odd reason we use everything but kebab-case in near-shell)

Reply:
> P.S. it is standard to have kebab-case naming for the CLI arguments (for some odd reason we use everything but kebab-case in near-shell)
 
@frol it's not immediately clear, but near-shell does support kebab-case arguments (translated automatically by `yargs`). 

as for having `experimental` prefix – my only concern is that it's not fun to type (especially in caps), not feeling super strongly. 

Also given outstanding proposal to redesign shell CLI, maybe it makes sense to use `near experimental validators`?  
Also does it really have to be "experimental"? Does it mean we can remove it or smth else?

Reply:
> it's not immediately clear, but near-shell does support kebab-case arguments 

:thinking: 

```
$ near --help
...
  --nodeUrl, --node_url      NEAR node URL  [string] [default: "http://localhost:3030"]
  --networkId, --network_id  NEAR network ID, allows using different keys based on network  [string] [default: "default"]
  --helperUrl                NEAR contract helper URL  [string]
  --keyPath                  Path to master account key  [string]
  --accountId, --account_id  Unique identifier for the account  [string]
```

Do you mean that even though the help message suggests camelCase and snake_case, it will be happy with kebab-case there as well?

> as for having experimental prefix – my only concern is that it's not fun to type (especially in caps), not feeling super strongly.

Well, we definitely don't want to keep it like that for ages, but it is a great way to allow users to use the latest API while allowing us to iterate quickly whenever we feel that changes are needed.

We cannot stabilize CLI feature before we stabilize the RPC endpoint. This implementation uses `EXPERIMENTAL_genesis_config` RPC, which is not settled due to the various changes to the genesis config structure (the most recent is staking storage, I believe).

> Also given outstanding proposal to redesign shell CLI, maybe it makes sense to use near experimental validators?

Looks good to me.

> Also does it really have to be "experimental"?

Well, given `EXPERIMENTAL_genesis_config` may introduce a breaking change, I consider this experimental. API stabilization is going to be a separate effort.

> Does it mean we can remove it or smth else?

Unfortunately, it does.

Reply:
> Do you mean that even though the help message suggests camelCase and snake_case, it will be happy with kebab-case there as well?

yes

> Well, given EXPERIMENTAL_genesis_config may introduce a breaking change, I consider this experimental. API stabilization is going to be a separate effort.

got it

Reply:
@vgrichina in my case experimental meant that there is no guarantee we won't change command. As you mentioned - we are redesigning command line interface, and I'm not 100% sure how this command should look ilke.